OpenRA.Game/GameRules/ActorInfo.cs

```diff
@@ -115,27 +115,34 @@ public IEnumerable<TraitInfo> TraitsInConstructOrder()
 			}).ToList();
 
 			var resolved = source.Where(s => s.Dependencies.Count == 0 && s.OptionalDependencies.Count == 0).ToList();
-			var unresolved = source.Except(resolved);
+			var unresolved = source.ToHashSet();
+			unresolved.ExceptWith(resolved);
 
-			var testResolve = new Func<Type, Type, bool>((a, b) => a == b || a.IsAssignableFrom(b));
+			bool TestResolve(Type a, Type b) => a.IsAssignableFrom(b);
```

Comment:
```suggestion
			bool TestResolve(Type a, Type b) => a == b || a.IsAssignableFrom(b);
```

why was this removed?

Reply:
`IsAssignableFrom` [does the same check](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Type.Helpers.cs,ba0cffea035fe210).

Reply:
ok